quantmsio/core/idxml.py

```diff
@@ -0,0 +1,521 @@
+        except (AttributeError, ValueError, TypeError):
+            pass
+
+        return additional_scores, q_value, posterior_error_probability
+
+    def _extract_meta_value(self, peptide_hit, key: str) -> Optional[float]:
+        """Extract and convert meta value to float"""
+        try:
+            value = peptide_hit.getMetaValue(key)
+            if value is not None:
+                return float(value)
+        except (AttributeError, ValueError, TypeError):
+            pass
+        return None
+
+    def _extract_additional_scores(self, peptide_hit) -> List[Dict]:
+        """Extract additional scores from peptide hit"""
+        additional_scores = []
+
+        try:
+            meta_keys = peptide_hit.getMetaValueKeys()
+            important_scores = [
+                "Luciphor_pep_score",
+                "Luciphor_global_flr",
+                "Luciphor_local_flr",
+                "consensus_support",
+                "search_engine_sequence",
+                "target_decoy",
+            ]
+
+            # Extract important scores first
+            for score_name in important_scores:
+                score_value = self._extract_meta_value(peptide_hit, score_name)
+                if score_value is not None:
+                    additional_scores.append(
+                        {"score_name": score_name, "score_value": score_value}
+                    )
+
+            # Extract other scores
+            excluded_scores = [
+                "q-value",
+                "Posterior Error Probability_score",
+            ] + important_scores
+            for key in meta_keys:
+                if key not in excluded_scores:
+                    score_value = self._extract_meta_value(peptide_hit, key)
+                    if score_value is not None:
+                        additional_scores.append(
+                            {"score_name": key, "score_value": score_value}
+                        )
+        except (AttributeError, ValueError, TypeError):
+            # Fallback to known scores only
+            known_scores = [
+                "Luciphor_pep_score",
+                "Luciphor_global_flr",
+                "Luciphor_local_flr",
+                "consensus_support",
+                "search_engine_sequence",
+                "target_decoy",
+            ]
+
+            for score_name in known_scores:
+                score_value = self._extract_meta_value(peptide_hit, score_name)
+                if score_value is not None:
+                    additional_scores.append(
+                        {"score_name": score_name, "score_value": score_value}
+                    )
+
+        return additional_scores
+
+    def _clean_sequence(self, sequence: str) -> str:
+        """Clean sequence by removing modification annotations"""
+        if "(" in sequence:
+            return re.sub(r"[\(\[].*?[\)\]]", "", sequence)
+        return sequence
+
+    def _parse_modifications(self, sequence: str) -> List[Dict]:
```

Comment:
A pyopenms AASequence object automatically has knowledge about all modifications. No need to use the sequence string to parse modifications again.